src/App.js

```diff
@@ -9,16 +9,27 @@ import RegisterPage from './pages/RegisterPage';
 import DashboardPage from './pages/DashboardPage';
 
 export default function App() {
-  const [isAuthenticated, setIsAuthenticated] = useState(true); 
+  const [isAuthenticated, setIsAuthenticated] = useState(false); 
 
   const setAuth = (valueToSet)=>{
     setIsAuthenticated(valueToSet);
   }
 
   const isAuth = async()=>{
     try {
-        //TODO
-        // verify is user is authenticated and set state by that
+       
+      const response = await fetch('http://localhost:5000/auth/verify', {
+        method: 'GET',
+        headers: {
+          token: localStorage.token
+        }
+      });
+
+      const parsedResponse = await response.json();
+
+      parsedResponse === true ? setIsAuthenticated(true) : setIsAuthenticated(false);
```

Comment:
Because parsedResponse is not always boolean type of variable, if there is no token, middleware authorization function is going to return res.json('Not Authorize'). parsedResponse is going to be string value of 'Not Authorize', and because in parsedResponse is some data, it is treated like true. So i concluded when parsedResponse is true and is boolean type (it's same by value and type) authentication is set to true, in all other cases is false.

Reply:
You are right but this is not good, I don't have idea right from my head but you should consider changing this, because it looks awful. I'm not that used to JS but in java or python I would have some kind of standardization to recognize this case of response.